miner/worker.go

```diff
@@ -786,7 +786,9 @@ func (w *worker) commitTransactions(txs *types.TransactionsByPriceAndNonce, coin
 	txCurr := &tx
 	//prefetch txs from all pending txs
 	txsPrefetch := txs.Copy()
-	w.prefetcher.PrefetchMining(txsPrefetch, w.current.header, w.current.gasPool.Gas(), w.current.state.Copy(), *w.chain.GetVMConfig(), interruptCh, txCurr)
+	newStatedb := w.current.state.Copy()
```

Comment:
`Copy` doesn't copy filed `writeOnSharedStorage`, would skip `sharedpool` in `prefetcher` routine

Reply:
since there're lots of references of `Copy`, could add an exported method `setSharedPool` to enable `writeOnSharedStorage` for specific scenarios.